### ISSUE TITLE
(RLENG) CMake error.

### ISSUE THREAD

**Issue by kdesnos** [preesm/spider2#1]
When attempting to build Spider2 with Cmake 3.13 on Linux (following README.md guidelines), the following error occurs:

```terminal
CMake Error at /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
```

`GTEST` does not seem to be standard package in these version of CMake. 
To fix this, the CMake module should be given or the requirement for a more recent version of CMake should be set in the `CMakeLists.txt` file.

**Comment by orlyn44** [comment]
There is an explaination in the test subfolder to fix this issue:
To install GTest: 
on Linux: 
sudo apt install libgtest-dev 
cd /usr/src/gtest sudo 
su cmake CMakeLists.txt
make -j$(nproc) 
cp *.a /usr/lib

However, I think you can still build the library without gtest, you won't be able to compile the test though.

**Comment by orlyn44** [comment]
Adding a reference to the method for fixing the issue in the main readme.

**Comment by kdesnos** [comment]
The guide in the README file will, by default, cause an error on most installation. I believe this is not a very good first impression when trying a new library.

To this issue, I see three options:

- Adding the `-DBUILD_TESTING=OFF` option to the `cmake` command in scripts. This way, everything works out of the box, and, it is obvious that something should be done to activate testing.
- Changing the root `CMakeLists.txt` to disable testing by default. The effect is the same as previous option, with the advantage of simplifying the command in the README compilation guide, but with the drawback of being less obvious for the developer that may want to try testing.
- Using the [official cmake-based auto-build technique for GoogleTest](https://github.com/google/googletest/tree/master/googletest#incorporating-into-an-existing-cmake-project).

**Comment by farresti** [comment]
fixed with #09372ec2b75c59b1bcfb06a05b767ad14fe0c068

**Comment by farresti** [comment]
fixed with https://github.com/preesm/spider2/commit/09372ec2b75c59b1bcfb06a05b767ad14fe0c068
